packages/jobs/lib/execution/sync.ts

```diff
@@ -92,6 +93,7 @@ export async function startSync(task: TaskSync, startScriptFn = startScript): Pr
                 meta: { scriptVersion: syncConfig.version }
             }
         );
+        logCtx.attachSpan(new OtlpSpan(logCtx.operation, startedAt));
```

Comment:
I am not a huge fan of this pseudo independent `attachSpan` function that we need to sprinkle all over the codebase. 
Since it always depends on a `logContextGetter.create` why not making it a part of it. 
can `logContextGetter.create` take a `otlpSpan?: { startedAt?: Date } params`. When it is passed the `create` function can set `span = new OtlpSpan(...)`

Reply:
`Since it always depends`, that's what I'm trying to partially remove actually, the problem with OTLP right now is that it assumes logCtx have the operation available at all times, which is not the case for syncs (and in the future in async actions).

When I need to recreate the span in `handleSuccess`, it's currently going through `logContextGetter.create` but it's only working because we are fetching the operation from ES. Because I want to get rid of that to optimize the code and I can't assume logCtx.operation will be there in the future.

Right now it's not fully fixed as my PR desc since I'm still using logCtx.operation, I still need to figure out a good solution